deploy/upgrade_db_server.yml

```diff
@@ -72,48 +174,73 @@
-                  --old-datadir {{ pp_old_datadir }}
-                  --new-datadir {{ pp_datadir }}
-                  --old-bindir {{ pp_prefix }}/{{ pp_old_install_path }}/bin
-                  --new-bindir {{ pp_prefix }}/{{ pp_install_path }}/bin
-                  --link
-                  -U {{ pp_serviceaccount }}
-                  --check
-      environment:
-        - PGPASSWORD: "{{ pp_superpassword }}"
+    - name: Drop temp scripts
+      postgresql_exec:
+        login_user: "{{ pp_superaccount }}"
+        login_password: "{{ pp_superpassword }}"
+        login_host: localhost
+        db: postgres
+        script: "{{ item }}"
+        port: "{{ pp_serverport }}"
+      with_items: "{{ drop_temp_scripts }}"
+      when: '"citus_coordinator" in group_names'
 
-    - name: Run pg_upgrade
-      #become: yes
-      #become_method: sudo
-      #become_user: "{{ pp_serviceaccount }}"
-      sudo: yes
-      remote_user: "{{ pp_serviceaccount }}"
-      command: >
-               {{ pp_prefix }}/{{ pp_install_path }}/bin/pg_upgrade
-                  --old-datadir {{ pp_old_datadir }}
-                  --new-datadir {{ pp_datadir }}
-                  --old-bindir {{ pp_prefix }}/{{ pp_old_install_path }}/bin
-                  --new-bindir {{ pp_prefix }}/{{ pp_install_path }}/bin
-                  --link
-                  -U {{ pp_serviceaccount }}
-      environment:
-        - PGPASSWORD: "{{ pp_superpassword }}"
+    - name: Restart sequence
+      postgresql_exec:
+        login_user: "{{ pp_superaccount }}"
+        login_password: "{{ pp_superpassword }}"
+        login_host: localhost
+        db: postgres
+        script: "{{ item }}"
+        port: "{{ pp_serverport }}"
+      with_items: "{{ restart_sequence_scripts }}"
+      when: '"citus_coordinator" in group_names'
+
+    - name: Create trigger
+      postgresql_exec:
+        login_user: "{{ pp_superaccount }}"
+        login_password: "{{ pp_superpassword }}"
+        login_host: localhost
+        db: postgres
+        script: "{{ create_trigger_script.script }}"
+        port: "{{ pp_serverport }}"
+      when: '"citus_coordinator" in group_names'
+
+    - name: Register trigger
+      postgresql_exec:
+        login_user: "{{ pp_superaccount }}"
+        login_password: "{{ pp_superpassword }}"
+        login_host: localhost
+        db: postgres
+        script: "{{ item }}"
+        port: "{{ pp_serverport }}"
+      with_items: "{{ register_trigger_scripts }}"
+      when: '"citus_coordinator" in group_names'
+
+    - name: Set dependency
+      postgresql_exec:
+        login_user: "{{ pp_superaccount }}"
+        login_password: "{{ pp_superpassword }}"
+        login_host: localhost
+        db: postgres
+        script: "{{ set_dependency_scripts.script }}"
+        port: "{{ pp_serverport }}"
+      when: '"citus_coordinator" in group_names'
 
-- hosts: db_server
+- hosts: all_db
   become: yes
   become_method: sudo
```

Comment:
@rioh Add `serial: 1` and `max_failure_percentage: 1%` as described here: http://docs.ansible.com/ansible/latest/playbooks_delegation.html#rolling-update-batch-size, and here: http://docs.ansible.com/ansible/latest/playbooks_delegation.html#maximum-failure-percentage, to ensure failure of any one host will stop the process.

Reply:
@rioh Can you add the serial part here too?  This role will restart the server(s), so we should do that in order rather than parallel..